provision/provision.go

```diff
@@ -240,6 +240,10 @@ type Provisioner interface {
 	RegisterUnit(Unit) error
 }
 
+type StartupMessageProvisioner interface {
```

Comment:
Could you rename it to `MessageProvisioner`? It's more generic and allow us to add other messages in the future.